stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/DOM.java

```diff
@@ -208,6 +214,97 @@ public void run() {
     });
   }
 
+  @ChromeDevtoolsMethod
+  public PerformSearchResponse performSearch(JsonRpcPeer peer, final JSONObject params) {
+    return mDOMProvider.postAndWait(new UncheckedCallable<PerformSearchResponse>() {
+      @Override
+      public PerformSearchResponse call() {
+        final PerformSearchRequest request = mObjectMapper.convertValue(
+            params,
+            PerformSearchRequest.class);
+
+        // Searches are case insensitive so convert
+        // the query to lower case for simpler comparisons.
+        final String lowerCaseQuery = request.query.toLowerCase();
+
+        final Object root = mShadowDOM.getRootElement();
+
+        // Each search action has a unique ID so that
+        // it can be queried later.
+        final String id = UUID.randomUUID().toString();
+
+        final ArrayList<Integer> results = new ArrayList<>();
+
+        findMatches(lowerCaseQuery, root, results);
+
+        mSearchResults.put(id, results);
+
+        final PerformSearchResponse response = new PerformSearchResponse();
+        response.searchId = id;
+        response.resultCount = results.size();
+
+        return response;
+      }
+    });
+  }
+
+  private void findMatches(String query, Object root, ArrayList<Integer> resultIds) {
+    final ElementInfo info = mShadowDOM.getElementInfo(root);
+
+    for (int i = 0, size = info.children.size(); i < size; i++) {
+      final Object element = info.children.get(i);
+
+      if (doesElementMatch(query, element)) {
+        resultIds.add(mObjectIdMapper.getIdForObject(element));
+      }
+
+      findMatches(query, element, resultIds);
+    }
+  }
+
+  private boolean doesElementMatch(final String query, final Object element) {
+      final NodeDescriptor descriptor = mDOMProvider.getNodeDescriptor(element);
+
+      final AttributeListAccumulator accumulator = getCachedAttributeListAccumulator();
+
+      try {
+        descriptor.getAttributes(element, accumulator);
```

Comment:
We'll need to update this code later, re: #228 that I just opened. Just a heads up :)